lib/sign-api-message.ts

```diff
@@ -0,0 +1,31 @@
+import { type SignTypedDataParameters } from "@wagmi/core";
+import { signTypedData } from "@wagmi/core";
+
+import { config as wagmiConfig } from "@/configs/wagmi";
+import { hypercertApiSigningDomain } from "@/configs/constants";
+
+export async function signMessage(
+  signerAddress: string,
+  chainId: number,
+  config: Pick<SignTypedDataParameters, "types" | "primaryType" | "message">,
+): Promise<`0x${string}`> {
+  if (!chainId) {
+    throw new Error("No chainId found");
+  }
+
+  // The type assertion to any is necessary because TS will otherwise complain,
+  // but the config is valid.
+  const signature = await signTypedData(wagmiConfig as any, {
```

Comment:
The config is generated using `createConfig`. But the return type is probably dynamically infered?

Could we use the `useConfig` hook and provide the returned config as input to the method? 

https://wagmi.sh/react/api/hooks/useConfig